File: TEST/DAY_30_1N_ALERTDIALOG/app/src/main/java/com/example/velic/day_30_1_alertdialog/MyDialogFragment8.java
package com.example.velic.day_30_1_alertdialog;

import android.app.Dialog;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;

import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;

public class MyDialogFragment8 extends DialogFragment {
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        LayoutInflater inflater = getActivity().getLayoutInflater();
        View view = inflater.inflate(R.layout.fragment_custom, null);
        builder.setView(view);
        // Остальной код
        return builder.create();
    }
}

File: TEST/DAY_30_1_ALERT_Dialog_kt/app/src/main/java/com/example/day_30_1_alert_dialog_k/MyDialogFragment8.kt
package com.example.day_30_1_alert_dialog_k

import android.app.Dialog
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View

import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
// AlertDialog с собственной разметкой
class MyDialogFragment8 : DialogFragment() {
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(activity!!)
        val inflater = activity!!.layoutInflater
        val view = inflater.inflate(R.layout.fragment_custom, null)
        builder.setView(view)
        // Остальной код
        return builder.create()
    }
}
